src/components/SnackChat/SnackChat.js

```diff
@@ -78,112 +59,140 @@ class SnackChat extends Component {
         return;
+      } else {
+        this.setState({itemPositions});
       }
 
-      // Video background
-      const video = this.webcam.current.webcam.current.video;
-      this.ctx.save();
-      this.drawBackground(video);
-
-      this.ctx.fillStyle = 'rgba(0.6, 0.6, 0.6, 0.6)';
-      this.ctx.fillRect(
-        0,
-        0,
-        this.canvas.current.width,
-        this.canvas.current.height
-      );
-      // draw items
-      this.ctx.save();
-      itemPositions.forEach(item => {
-        this.ctx.translate(item.x, item.y);
-        this.ctx.rotate(item.rotation);
-        this.ctx.drawImage(
-          this.filter,
-          -0.1 * this.canvas.current.width,
-          -0.1 * this.canvas.current.width,
-          FALLING_SNACK_SIZE * this.canvas.current.width,
-          FALLING_SNACK_SIZE * this.canvas.current.height
-        );
-        this.ctx.setTransform(1, 0, 0, 1, 0, 0);
-      });
-      this.ctx.restore();
-
       // update positions
       itemPositions.forEach(item => {
         item.y =
-          ((FALLING_SNACK_SIZE * this.canvas.current.height + item.y + 3) %
-            ((1 + 2 * FALLING_SNACK_SIZE) * this.canvas.current.height)) -
-          FALLING_SNACK_SIZE * this.canvas.current.height;
+          ((FALLING_SNACK_SIZE * FEED_SIZE + item.y + 3) %
+            ((1 + 2 * FALLING_SNACK_SIZE) * FEED_SIZE)) -
+          FALLING_SNACK_SIZE * FEED_SIZE;
         item.rotation = (item.rotation + 0.03) % (Math.PI * 2);
       });
-
       requestAnimationFrame(drawFallingSnacks);
     };
-
     requestAnimationFrame(drawFallingSnacks);
-    this.ctx.clearRect(
-      0,
-      0,
-      this.canvas.current.width,
-      this.canvas.current.height
+  };
+
+  clipEllipse(ctx, centerX, centerY, width, height) {
+    ctx.beginPath();
+    ctx.moveTo(centerX, centerY - height / 2);
+    ctx.bezierCurveTo(
+      centerX + width / 2,
+      centerY - height / 2,
+      centerX + width / 2,
+      centerY + height / 2,
+      centerX,
+      centerY + height / 2
+    );
+    ctx.bezierCurveTo(
+      centerX - width / 2,
+      centerY + height / 2,
+      centerX - width / 2,
+      centerY - height / 2,
+      centerX,
+      centerY - height / 2
+    );
+    ctx.clip();
+  }
+
+  drawBackground = video => {
+    this.ctx.scale(-CANVAS_SCALE, CANVAS_SCALE);
+    this.ctx.drawImage(
+      video,
+      Math.abs(video.videoWidth - this.canvas.current.width) / 2 -
+        video.videoWidth,
+      0
     );
+    this.ctx.restore();
+  };
+
+  generateSnackchat = async () => {
+    const video = this.webcam.current.webcam.current.video;
+
+    this.ctx.save();
+    this.drawBackground(video);
+
+    const filter = new Image();
+    filter.src = this.filter;
+
+    const shoulders = this.state.averageBodyPosition.shoulders;
+    this.ctx.save();
+    this.ctx.drawImage(
+      filter,
+      shoulders.rightX -
+        shoulders.span * 1.5 +
+        shoulders.span * shoulders.angle,
```

Comment:
Would be good to extract these again

```
dx = ...
dy = ...
dWidth = ...
dHeight = ...
this.cxt.drawImage(filter, dx, dx, dWidth, dHeight)
```

Just makes it clear and don't have to hunt for the docs every time we read the code